core/audit_logger.py

```
@@ -0,0 +1,240 @@
+        handler.setFormatter(formatter)
+
+        self.logger.addHandler(handler)
+
+    def log_event(
+        self,
+        event_type: AuditEventType,
+        user: str = "unknown",
+        ip: str = "unknown",
+        resource: str = "",
+        action: str = "",
+        details: Dict[str, Any] = None,
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)